sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -156,6 +156,12 @@ abstract class Expression extends TreeNode[Expression] {
 
   private def reduceCodeSize(ctx: CodegenContext, eval: ExprCode): Unit = {
     // TODO: support whole stage codegen too
+    //
+    // NOTE: We could use `CodeGenerator.defineSingleSplitFunction` here for the code path
```

Comment:
Why add this comment?

Reply:
To avoid duplicate work. Currently, this func and `defineSingleSplitFunction` has too similar logic.